client/src/main/java/cloud/prefab/client/internal/ConfigClientImpl.java

```diff
@@ -323,6 +345,56 @@ public void onCompleted() {
       );
   }
 
+  private void startLogStatsUploadExecutor() {
+    ScheduledExecutorService scheduledExecutorService = Executors.newScheduledThreadPool(
```

Comment:
I wonder if we should make this class closeable so that we can shut down this and the checkpoint executor? Or maybe just use `MoreExecutors#getExitingExecutorService` like we do for streaming

Reply:
i switched to getExiting for both, I do like the idea of making the class closeable so I'm going to park that on our trello board